src/shared_modules/indexer_connector/testtool/main.cpp

```diff
@@ -1,40 +1,152 @@
+    return ss.str();
+}
+
+nlohmann::json fillWithRandomData(const nlohmann::json& templateJson)
+{
+    nlohmann::json result;
+
+    for (auto& [key, value] : templateJson.items())
     {
-        "type": "test-index",
-        "data": {
-            "message": "Hello world!"
-        },
-        "id": "1",
-        "operation": ""
-    })");
-
-    indexerConnector.publish(R"(
+        if (value.is_object())
+        {
+            if (key == "properties")
+            {
+                result.merge_patch(fillWithRandomData(value));
+            }
+            else
+            {
+                result[key] = fillWithRandomData(value);
+            }
+        }
+        else if (key == "type")
+        {
+            if (value.get<std::string>() == "keyword")
+            {
+                result = generateRandomString(10);
+            }
+            else if (value.get<std::string>() == "long")
+            {
+                result = generateRandomInt(0, 1000);
+            }
+            else if (value.get<std::string>() == "float")
+            {
+                result = generateRandomFloat(0.0, 100.0);
+            }
+            else if (value.get<std::string>() == "date")
+            {
+                result = generateTimestamp();
+            }
+        }
+    }
+
+    return result;
+}
+
+int main(const int argc, const char* argv[])
+{
+    try
+    {
+        CmdLineArgs cmdArgParser(argc, argv);
+
+        // Read configuration file.
+        std::ifstream configurationFile(cmdArgParser.getConfigurationFilePath());
+        if (!configurationFile.is_open())
+        {
+            throw std::runtime_error("Could not open configuration file.");
+        }
+        nlohmann::json configuration;
+        configurationFile >> configuration;
+
+        // Create indexer connector.
+        IndexerConnector indexerConnector(configuration, cmdArgParser.getTemplateFilePath());
+        // Read events file.
+        // If the events file path is empty, then the events are generated
+        // automatically.
+        if (!cmdArgParser.getEventsFilePath().empty())
+        {
+            std::ifstream eventsFile(cmdArgParser.getEventsFilePath());
+            if (!eventsFile.is_open())
+            {
+                throw std::runtime_error("Could not open events file.");
+            }
+            nlohmann::json events;
+            eventsFile >> events;
+
+            indexerConnector.publish(events.dump());
+        }
+        else if (cmdArgParser.getAutoGenerated())
+        {
+            srand(static_cast<unsigned int>(time(nullptr)));
+            const auto n = cmdArgParser.getNumberOfEvents();
+            // Read template file.
+
+            std::ifstream templateFile(cmdArgParser.getTemplateFilePath());
+            if (!templateFile.is_open())
+            {
+                throw std::runtime_error("Could not open template file.");
+            }
+
+            nlohmann::json templateData;
```

Comment:
Coding style: the `nlohmann` variable can be initialized directly from the `std::ifstream` object.